src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Extensions/ApplicationInsightsServiceCollectionExtensions.cs

```diff
@@ -159,10 +159,16 @@ public static IServiceCollection AddApplicationInsights(this IServiceCollection
                 // role environment telemetry initializer is needed to correlate heartbeats to particular host
 
                 var activeConfig = TelemetryConfiguration.Active;
-                if (!string.IsNullOrEmpty(options.InstrumentationKey) &&
-                string.IsNullOrEmpty(activeConfig.InstrumentationKey))
+                if (string.IsNullOrEmpty(activeConfig.InstrumentationKey))
                 {
-                    activeConfig.InstrumentationKey = options.InstrumentationKey;
+                    if (!string.IsNullOrEmpty(options.ConnectionString))
+                    {
+                        activeConfig.ConnectionString = options.ConnectionString;
+                    }
+                    else if (!string.IsNullOrEmpty(options.InstrumentationKey))
+                    {
+                        activeConfig.InstrumentationKey = options.InstrumentationKey;
+                    }
                 }
```

Comment:
The logic here seems off. I think we want to set active ConnectionString (if not already set AND it's set in options). If that doesn't happen, then do the same thing with InstrumentationKey.

Basically, we need to get the active.ConnectionString or InstrumentationKey to match what our options say.

Reply:
Ah, good point. I think my reasoning was I wanted to make sure InstrumentationKey doesn't overwrite the value that ConnectionString might set if both are present, but as long as ConnectionString is set last it should be fine.